src/worker_job.cc

```diff
@@ -175,6 +187,25 @@ namespace detail {
 		if(m_future.wait_for(std::chrono::seconds(0)) == std::future_status::ready) {
 			m_buffer_mngr.unlock(pkg.cid);
 
+#if CELERITY_ACCESSOR_BOUNDARY_CHECK
+			const auto data = std::get<execution_data>(pkg.data);
+			auto tsk = m_task_mngr.get_task(data.tid);
+
+			for(size_t i = 0; i < m_oob_indices_per_accessor.size(); ++i) {
+				const id<3>& oob_min = m_oob_indices_per_accessor[i][0];
+				const id<3>& oob_max = m_oob_indices_per_accessor[i][1];
+
+				if(oob_max != id<3>{1, 1, 1}) {
+					const auto& access_map = tsk->get_buffer_access_map();
+					const auto acc_sr =
+					    grid_box_to_subrange(access_map.get_requirements_for_nth_access(i, tsk->get_dimensions(), data.sr, tsk->get_global_size()));
+					const auto oob_sr = subrange<3>(oob_min, range_cast<3>(oob_max - oob_min));
+					CELERITY_ERROR("Out-of-bounds access in host kernel detected: Accessor {} for buffer {} attempted to access indices between {} which are "
+					               "outside of mapped subrange {}", i, access_map.get_nth_access(i).first, oob_sr, acc_sr);
```

Comment:
We might want to use the buffer name here if one is available?
(this would also apply to device accesses)